book/animations.md

```diff
@@ -638,226 +634,329 @@ trigger it once every 2 seconds:
 your browser; [here](examples/example13-width-transition.html) is the width
 animation example)
 
-Implement this CSS property. The strategy will be almost the same as in
-JavaScript: define an object on which we can call `animate` on each
-animation frame, causing the animation to advance one frame. And when it's
-done, remove it. There can be more than one CSS transition, so store them
-in a 2D `animations` dictionary keyed by node and CSS property
-name.[^delete-complicated]
+Implement this CSS property. The strategy will be almost the same as
+in JavaScript: define an object on which we can call `animate` on each
+animation frame, causing the animation to advance one frame. Multiple
+elements can animate at a time, so let's store animations in an
+`animations` dictionary on each node, keyed by the property being
+animated:[^delete-complicated]
 
-[^delete-complicated]: And because we're iterating over a dictionary, we can't
-delete entries right then. Instead, we have to save off a list of entries to
-delete and then loop again to delete them. That's why there are two loops and
-the `to_be_deleted` list.
+[^delete-complicated]: For simplicity, this code leaves animations in
+    the `animations` dictionary even when they're done animating.
+    Removing them would be necessary, however, for really long-running
+    tabs where just looping over all the already-completed animations
+    can take a while.
 
+<!--
+This is not actually present in the code; it's hacked in with a
```

Comment:
Ah—this comment refers to the fact that the _reader_ should add the `style` and `animations` fields to the `Element` and `Text` classes, but I didn't because it would require copying in most of the browser code from previous chapters before editing it. I think we should just do that, it was a dumb expedient.

Reply:
Go it, ok. Acctually, now that you explained it, I'm pretty sure I tried exactly this refactor two months ago and abandoned it for precisely this reason. Do you think moving the dictionaries onto the nodes themselves adds enough value, or should we revert back to storing a double dictionary on the Tab?